internal/provider/resource_managed_identity_access_rules.go

```diff
@@ -0,0 +1,355 @@
+	}
+
+	// Copy fields returned by Tharsis to the plan.  Apparently, must copy all fields, not just the computed fields.
+	plan.RunStage = types.StringValue(string(updated.RunStage))
+
+	plan.AllowedUsers = []types.String{}
+	for _, user := range updated.AllowedUsers {
+		plan.AllowedUsers = append(plan.AllowedUsers, types.StringValue(user.Username))
+	}
+
+	plan.AllowedServiceAccounts = []types.String{}
+	for _, serviceAccount := range updated.AllowedServiceAccounts {
+		plan.AllowedServiceAccounts = append(plan.AllowedServiceAccounts, types.StringValue(serviceAccount.ResourcePath))
+	}
+
+	plan.AllowedTeams = []types.String{}
+	for _, team := range updated.AllowedTeams {
+		plan.AllowedTeams = append(plan.AllowedTeams, types.StringValue(team.Name))
+	}
+
+	// Set the response state to the fully-populated plan.
+	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (t *managedIdentityAccessRuleResource) Delete(ctx context.Context,
+	req resource.DeleteRequest, resp *resource.DeleteResponse) {
+
+	// Get the current state.
+	var state ManagedIdentityAccessRuleModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Delete the access rule via Tharsis.
+	// The ID is used to find the record to delete.
+	err := t.client.ManagedIdentity.DeleteManagedIdentityAccessRule(ctx,
+		&ttypes.DeleteManagedIdentityAccessRuleInput{
+			ID: state.ID.ValueString(),
+		})
+	if err != nil {
+
+		// Handle the case that the access rule no longer exists.
+		if t.isErrorRuleNotFound(err) {
+			resp.State.RemoveResource(ctx)
+			return
+		}
+
+		resp.Diagnostics.AddError(
+			"Error deleting managed identity access rule",
+			err.Error(),
+		)
+		return
+	}
+}
+
+// ImportState helps the provider implement the ResourceWithImportState interface.
+func (t *managedIdentityAccessRuleResource) ImportState(ctx context.Context,
+	req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+
+	// Retrieve import ID and save to id attribute
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+
+	// Setting of the ManagedIdentityID field during import is handled in the Read method.
+
+}
+
+// valueStrings converts a slice of types.String to a slice of strings.
+func (t *managedIdentityAccessRuleResource) valueStrings(arg []types.String) []string {
+	result := make([]string, len(arg))
+	for ix, bigSValue := range arg {
+		result[ix] = bigSValue.ValueString()
+	}
+	return result
+}
+
+// isErrorRuleNotFound returns true iff the error message is that an access rule was not found.
+// Don't check the ID, because the available ID is the global id, while the ID in the message is a local ID.
+// In theory, we should never see a message that some other ID was not found.
+func (t *managedIdentityAccessRuleResource) isErrorRuleNotFound(e error) bool {
+	// Omission of the leading 'M' is intentional in case the SDK changes to lowercase.
```

Comment:
We'll need to add a follow-on story to improve the errors that are returned by the SDK to avoid having to compare string values.